Source/WebCore/svg/SVGPathByteStream.h

```diff
@@ -39,71 +40,158 @@ namespace WebCore {
+
+        using Bytes = Vector<uint8_t>;
+
+        static Ref<Data> create(Bytes&& bytes)
+        {
+            return adoptRef(*new Data(WTFMove(bytes)));
+        }
+
+        void updatePath(const Path& path) const
+        {
+            m_path = path;
+        }
+
+        static Ref<Data> empty()
+        {
+            static MainThreadNeverDestroyed<Data> singleton;
+            return Ref { singleton.get() };
+        }
+
+        const Bytes& bytes() const { return m_bytes; }
+
+        void append(uint8_t byte)
+        {
+            m_path = { };
+            m_bytes.append(byte);
+        }
+
+        void append(std::span<const uint8_t> bytes)
+        {
+            m_path = { };
+            m_bytes.append(bytes);
+        }
+
+        void append(const Bytes& other)
+        {
+            m_path = { };
+            m_bytes.appendVector(other);
+        }
+
+        const Path& path() const { return m_path; }
+
+        Ref<Data> copy() const
+        {
+            return adoptRef(*new Data(*this));
+        }
+
+        friend bool operator==(const Data& a, const Data& b)
+        {
+            return a.m_bytes == b.m_bytes;
+        }
+
+        unsigned size() const { return m_bytes.size(); }
+        bool isEmpty() const { return m_bytes.isEmpty(); }
+
+    private:
+        Data(Bytes&& bytes)
+            : m_bytes(WTFMove(bytes))
+        {
+        }
+
+        Data() = default;
+        Data(const Data& data)
+            : m_bytes(data.m_bytes)
+            , m_path(data.m_path)
+        {
+        }
+
+        Bytes m_bytes;
+        mutable Path m_path;
+    };
+
+    using DataIterator = Data::Bytes::const_iterator;
+
+    SVGPathByteStream()
+        : m_data(Data::empty())
+    {
+    }
 
     SVGPathByteStream(StringView string)
+        : SVGPathByteStream()
     {
         buildSVGPathByteStreamFromString(string, *this, UnalteredParsing);
     }
 
     SVGPathByteStream(const SVGPathByteStream& other)
-        : CanMakeSingleThreadWeakPtr<SVGPathByteStream>()
+        : SVGPathByteStream()
     {
         *this = other;
     }
 
     SVGPathByteStream(SVGPathByteStream&& other)
+        : SVGPathByteStream()
     {
         *this = WTFMove(other);
     }
 
-    SVGPathByteStream(Data&& data)
+    SVGPathByteStream(Ref<Data>&& data)
         : m_data(WTFMove(data))
     {
     }
 
-    SVGPathByteStream& operator=(const SVGPathByteStream& other)
+    SVGPathByteStream(Data::Bytes&& data)
+        : m_data(Data::create(WTFMove(data)))
     {
-        if (*this == other)
-            return *this;
-        m_data = other.m_data;
-        return *this;
     }
 
-    SVGPathByteStream& operator=(SVGPathByteStream&& other)
+    SVGPathByteStream& operator=(const SVGPathByteStream& other) = default;
+    SVGPathByteStream& operator=(SVGPathByteStream&& other) = default;
+
+    friend bool operator==(const SVGPathByteStream& a, const SVGPathByteStream& b)
```

Comment:
`= default` would work it seems ?

Reply:
Right! Changed.

Reply:
Ah, we need it since we do not want to do `==` for CanSingleThreadWeakPtr's fields.